pkg/console/controllers/oauthclients/oauthclients.go

```diff
@@ -193,9 +195,12 @@ func (c *oauthClientsController) sync(ctx context.Context, controllerContext fac
 		syncErr = c.syncAuthTypeOIDC(ctx, controllerContext, operatorConfig, authnConfig)
 	}
 
-	// AuthStatusHandler manages fields that are behind the CustomNoUpgrade and TechPreviewNoUpgrade featuregate sets
-	// call Apply() only if they are enabled, otherwise server-side apply will complain
-	if featureGates.Spec.FeatureSet == configv1.TechPreviewNoUpgrade || featureGates.Spec.FeatureSet == configv1.CustomNoUpgrade {
+	oidcClientsSchema, err := authnConfigHasOIDCFields(c.crdLister)
+	if err != nil {
+		return err
+	}
+
+	if oidcClientsSchema {
 		if err := c.authStatusHandler.Apply(ctx, authnConfig); err != nil {
```

Comment:
why can't this just move into the block with `configv1.AuthenticationTypeOIDC` a few lines up?